### PR TITLE
Week 2-2 Ifs, loops and functions : loops-and-functions branch 

### PR DESCRIPTION
**Pull request by cavblk**
Work with : 
- ifs 
- loops
- functions
- tried to avoid error catching if not integer via checking if character entered can be part of an integer; first char only can also be - 
It was done before memory savers, perhaps it can be revisited

### CODE REVIEW

File: main_hw2.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    value = input()
+    # Avoid catch exception, check if all chars are digits ; LE: could have been done with isdigit ? :)
+    if value[0] not in chiffres and value[0] not in {'-'}:  # First char can be (-) as well
+        # print(value[0], 'not in range')
+        is_int = False
+    for i in range(1, len(value)):
+        # print(value[i])
+        if value[i] not in chiffres:
+            # print(value[i], 'not in range')
+            is_int = False
```
Comment (mihaivladu) [inline main_hw2.py:84]: You have to `break` the for loop here once you find a character that is not a digit. Please check what happens if you type `abc1` from the keyboard.

File: main_hw2.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    for value in kwargs.values():
+        # print(value)
+        type_value = type(value).__name__
+        if type_value == 'float' or type_value == 'int':
+            r_s += value
+
+    return r_s
+
+
+def sum_rec(n):
```
Comment (mihaivladu) [inline main_hw2.py:19]: Please refactor `sum_rec,` `sum_rec_even` and `sum_rec_odd` as one function.

**Review by mihaivladu — CHANGES_REQUESTED** [review]

File: main_hw2.py (hunk 1 above)
Comment (cavblk) [reply main_hw2.py:84]: Please enter an int:

abc1
0

Comment (mihaivladu) [reply main_hw2.py:84]: Right. Please check with `1abc1` :)

Comment (cavblk) [reply main_hw2.py:84]: Please enter an int:

1abc1
0

File: main_hw2.py (hunk 2 above)
Comment (cavblk) [reply main_hw2.py:19]: created sum_rec_refact which returns tuple (sum, sum_even) sum_odd is calculated as difference between them; or one can put it in returned tuple as well

**Pull request merged**